libcore.io.GaiException  libcore.io.Posix.getaddrinfo

```

io.reactivex.exceptions.OnErrorNotImplementedException: 
  at io.reactivex.internal.observers.EmptyCompletableObserver.dispose (EmptyCompletableObserver.java)
  or                     .onError (EmptyCompletableObserver.java)
  or                     .onSubscribe (EmptyCompletableObserver.java)
  at io.reactivex.internal.operators.completable.CompletableFromSingle$CompletableFromSingleObserver.onError (CompletableFromSingle.java)
  or                     .onSubscribe (CompletableFromSingle.java)
  or                     .onSuccess (CompletableFromSingle.java)
  at io.reactivex.internal.operators.single.SingleObserveOn$ObserveOnSingleObserver.run (SingleObserveOn.java)
  at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run (HandlerScheduler.java)
  at android.os.Handler.handleCallback (Handler.java:733)
  at android.os.Handler.dispatchMessage (Handler.java:95)
  at android.os.Looper.loop (Looper.java:136)
  at android.app.ActivityThread.main (ActivityThread.java:5584)
  at java.lang.reflect.Method.invokeNative (Method.java)
  at java.lang.reflect.Method.invoke (Method.java:515)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run (ZygoteInit.java:1268)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1084)
  at dalvik.system.NativeStart.main (NativeStart.java)
Caused by: java.net.UnknownHostException: 
  at java.net.InetAddress.lookupHostByName (InetAddress.java:424)
  at java.net.InetAddress.getAllByNameImpl (InetAddress.java:236)
  at java.net.InetAddress.getAllByName (InetAddress.java:214)
  at okhttp3.Dns$1.lookup (Dns.java)
  at okhttp3.internal.connection.RouteSelector.connectFailed (RouteSelector.java)
  or                     .getHostString (RouteSelector.java)
  or                     .hasNext (RouteSelector.java)
  or                     .resetNextInetSocketAddress (RouteSelector.java)
  or                     .resetNextProxy (RouteSelector.java)
  at okhttp3.internal.connection.RouteSelector.nextProxy (RouteSelector.java)
  at okhttp3.internal.connection.RouteSelector.next (RouteSelector.java)
  at okhttp3.internal.connection.StreamAllocation.acquire (StreamAllocation.java)
  or                     .codec (StreamAllocation.java)
  or                     .deallocate (StreamAllocation.java)
  or                     .findConnection (StreamAllocation.java)
  or                     .findHealthyConnection (StreamAllocation.java)
  or                     .newStream (StreamAllocation.java)
  or                     .releaseAndAcquire (StreamAllocation.java)
  or                     .streamFailed (StreamAllocation.java)
  or                     .streamFinished (StreamAllocation.java)
  at okhttp3.internal.connection.StreamAllocation.acquire (StreamAllocation.java)
  or                     .codec (StreamAllocation.java)
  or                     .deallocate (StreamAllocation.java)
  or                     .findConnection (StreamAllocation.java)
  or                     .findHealthyConnection (StreamAllocation.java)
  or                     .newStream (StreamAllocation.java)
  or                     .releaseAndAcquire (StreamAllocation.java)
  or                     .streamFailed (StreamAllocation.java)
  or                     .streamFinished (StreamAllocation.java)
  at okhttp3.internal.connection.StreamAllocation.acquire (StreamAllocation.java)
  or                     .codec (StreamAllocation.java)
  or                     .deallocate (StreamAllocation.java)
  or                     .findConnection (StreamAllocation.java)
  or                     .findHealthyConnection (StreamAllocation.java)
  or                     .newStream (StreamAllocation.java)
  or                     .releaseAndAcquire (StreamAllocation.java)
  or                     .streamFailed (StreamAllocation.java)
  or                     .streamFinished (StreamAllocation.java)
  at okhttp3.internal.connection.ConnectInterceptor.intercept (ConnectInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.cache.CacheInterceptor.cacheWritingResponse (CacheInterceptor.java)
  or                     .combine (CacheInterceptor.java)
  or                     .intercept (CacheInterceptor.java)
  or                     .isEndToEnd (CacheInterceptor.java)
  or                     .stripBody (CacheInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.BridgeInterceptor.cookieHeader (BridgeInterceptor.java)
  or                     .intercept (BridgeInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RetryAndFollowUpInterceptor.cancel (RetryAndFollowUpInterceptor.java)
  or                     .createAddress (RetryAndFollowUpInterceptor.java)
  or                     .followUpRequest (RetryAndFollowUpInterceptor.java)
  or                     .intercept (RetryAndFollowUpInterceptor.java)
  or                     .isRecoverable (RetryAndFollowUpInterceptor.java)
  or                     .recover (RetryAndFollowUpInterceptor.java)
  or                     .sameConnection (RetryAndFollowUpInterceptor.java)
  or                     .setCallStackTrace (RetryAndFollowUpInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.logging.HttpLoggingInterceptor.bodyEncoded (HttpLoggingInterceptor.java)
  or                     .intercept (HttpLoggingInterceptor.java)
  or                     .isPlaintext (HttpLoggingInterceptor.java)
  or                     .setLevel (HttpLoggingInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.RealCall.getResponseWithInterceptorChain (RealCall.java)
  at okhttp3.RealCall.access$000 (RealCall.java)
  or                 .enqueue (RealCall.java)
  or                 .execute (RealCall.java)
  or                 .newRealCall (RealCall.java)
  at retrofit2.OkHttpCall.enqueue (OkHttpCall.java)
  or                     .execute (OkHttpCall.java)
  or                     .parseResponse (OkHttpCall.java)
  at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual (CallExecuteObservable.java)
  at io.reactivex.Observable.bufferSize (Observable.java)
  or                     .filter (Observable.java)
  or                     .flatMapSingle (Observable.java)
  or                     .fromArray (Observable.java)
  or                     .just (Observable.java)
  or                     .subscribe (Observable.java)
  or                     .subscribeOn (Observable.java)
  or                     .toFlowable (Observable.java)
  or                     .toList (Observable.java)
  at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual (BodyObservable.java)
  at io.reactivex.Observable.bufferSize (Observable.java)
  or                     .filter (Observable.java)
  or                     .flatMapSingle (Observable.java)
  or                     .fromArray (Observable.java)
  or                     .just (Observable.java)
  or                     .subscribe (Observable.java)
  or                     .subscribeOn (Observable.java)
  or                     .toFlowable (Observable.java)
  or                     .toList (Observable.java)
  at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual (ObservableSingleSingle.java)
  at io.reactivex.Single.doOnSubscribe (Single.java)
  or                    .error (Single.java)
  or                    .flatMap (Single.java)
  or                    .just (Single.java)
  or                    .observeOn (Single.java)
  or                    .subscribe (Single.java)
  or                    .zip (Single.java)
  or                    .zipArray (Single.java)
  at io.reactivex.internal.operators.single.SingleMap.subscribeActual (SingleMap.java)
  at io.reactivex.Single.doOnSubscribe (Single.java)
  or                    .error (Single.java)
  or                    .flatMap (Single.java)
  or                    .just (Single.java)
  or                    .observeOn (Single.java)
  or                    .subscribe (Single.java)
  or                    .zip (Single.java)
  or                    .zipArray (Single.java)
  at io.reactivex.internal.operators.single.SingleZipArray.subscribeActual (SingleZipArray.java)
  at io.reactivex.Single.doOnSubscribe (Single.java)
  or                    .error (Single.java)
  or                    .flatMap (Single.java)
  or                    .just (Single.java)
  or                    .observeOn (Single.java)
  or                    .subscribe (Single.java)
  or                    .zip (Single.java)
  or                    .zipArray (Single.java)
  at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run (SingleSubscribeOn.java)
  at io.reactivex.Scheduler$DisposeTask.run (Scheduler.java)
  at io.reactivex.internal.schedulers.ScheduledRunnable.run (ScheduledRunnable.java)
  at io.reactivex.internal.schedulers.ScheduledRunnable.call (ScheduledRunnable.java)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201 (ScheduledThreadPoolExecutor.java:152)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:265)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
  at java.lang.Thread.run (Thread.java:841)
Caused by: libcore.io.GaiException: 
  at libcore.io.Posix.getaddrinfo (Posix.java)
  at libcore.io.ForwardingOs.getaddrinfo (ForwardingOs.java:61)
  at java.net.InetAddress.lookupHostByName (InetAddress.java:405)
  at java.net.InetAddress.getAllByNameImpl (InetAddress.java:236)
  at java.net.InetAddress.getAllByName (InetAddress.java:214)
  at okhttp3.Dns$1.lookup (Dns.java)
  at okhttp3.internal.connection.RouteSelector.connectFailed (RouteSelector.java)
  or                     .getHostString (RouteSelector.java)
  or                     .hasNext (RouteSelector.java)
  or                     .resetNextInetSocketAddress (RouteSelector.java)
  or                     .resetNextProxy (RouteSelector.java)
  at okhttp3.internal.connection.RouteSelector.nextProxy (RouteSelector.java)
  at okhttp3.internal.connection.RouteSelector.next (RouteSelector.java)
  at okhttp3.internal.connection.StreamAllocation.acquire (StreamAllocation.java)
  or                     .codec (StreamAllocation.java)
  or                     .deallocate (StreamAllocation.java)
  or                     .findConnection (StreamAllocation.java)
  or                     .findHealthyConnection (StreamAllocation.java)
  or                     .newStream (StreamAllocation.java)
  or                     .releaseAndAcquire (StreamAllocation.java)
  or                     .streamFailed (StreamAllocation.java)
  or                     .streamFinished (StreamAllocation.java)
  at okhttp3.internal.connection.StreamAllocation.acquire (StreamAllocation.java)
  or                     .codec (StreamAllocation.java)
  or                     .deallocate (StreamAllocation.java)
  or                     .findConnection (StreamAllocation.java)
  or                     .findHealthyConnection (StreamAllocation.java)
  or                     .newStream (StreamAllocation.java)
  or                     .releaseAndAcquire (StreamAllocation.java)
  or                     .streamFailed (StreamAllocation.java)
  or                     .streamFinished (StreamAllocation.java)
  at okhttp3.internal.connection.StreamAllocation.acquire (StreamAllocation.java)
  or                     .codec (StreamAllocation.java)
  or                     .deallocate (StreamAllocation.java)
  or                     .findConnection (StreamAllocation.java)
  or                     .findHealthyConnection (StreamAllocation.java)
  or                     .newStream (StreamAllocation.java)
  or                     .releaseAndAcquire (StreamAllocation.java)
  or                     .streamFailed (StreamAllocation.java)
  or                     .streamFinished (StreamAllocation.java)
  at okhttp3.internal.connection.ConnectInterceptor.intercept (ConnectInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.cache.CacheInterceptor.cacheWritingResponse (CacheInterceptor.java)
  or                     .combine (CacheInterceptor.java)
  or                     .intercept (CacheInterceptor.java)
  or                     .isEndToEnd (CacheInterceptor.java)
  or                     .stripBody (CacheInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.BridgeInterceptor.cookieHeader (BridgeInterceptor.java)
  or                     .intercept (BridgeInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RetryAndFollowUpInterceptor.cancel (RetryAndFollowUpInterceptor.java)
  or                     .createAddress (RetryAndFollowUpInterceptor.java)
  or                     .followUpRequest (RetryAndFollowUpInterceptor.java)
  or                     .intercept (RetryAndFollowUpInterceptor.java)
  or                     .isRecoverable (RetryAndFollowUpInterceptor.java)
  or                     .recover (RetryAndFollowUpInterceptor.java)
  or                     .sameConnection (RetryAndFollowUpInterceptor.java)
  or                     .setCallStackTrace (RetryAndFollowUpInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.logging.HttpLoggingInterceptor.bodyEncoded (HttpLoggingInterceptor.java)
  or                     .intercept (HttpLoggingInterceptor.java)
  or                     .isPlaintext (HttpLoggingInterceptor.java)
  or                     .setLevel (HttpLoggingInterceptor.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.internal.http.RealInterceptorChain.proceed (RealInterceptorChain.java)
  or                     .request (RealInterceptorChain.java)
  at okhttp3.RealCall.getResponseWithInterceptorChain (RealCall.java)
  at okhttp3.RealCall.access$000 (RealCall.java)
  or                 .enqueue (RealCall.java)
  or                 .execute (RealCall.java)
  or                 .newRealCall (RealCall.java)
  at retrofit2.OkHttpCall.enqueue (OkHttpCall.java)
  or                     .execute (OkHttpCall.java)
  or                     .parseResponse (OkHttpCall.java)
  at retrofit2.adapter.rxjava2.CallExecuteObservable.subscribeActual (CallExecuteObservable.java)
  at io.reactivex.Observable.bufferSize (Observable.java)
  or                     .filter (Observable.java)
  or                     .flatMapSingle (Observable.java)
  or                     .fromArray (Observable.java)
  or                     .just (Observable.java)
  or                     .subscribe (Observable.java)
  or                     .subscribeOn (Observable.java)
  or                     .toFlowable (Observable.java)
  or                     .toList (Observable.java)
  at retrofit2.adapter.rxjava2.BodyObservable.subscribeActual (BodyObservable.java)
  at io.reactivex.Observable.bufferSize (Observable.java)
  or                     .filter (Observable.java)
  or                     .flatMapSingle (Observable.java)
  or                     .fromArray (Observable.java)
  or                     .just (Observable.java)
  or                     .subscribe (Observable.java)
  or                     .subscribeOn (Observable.java)
  or                     .toFlowable (Observable.java)
  or                     .toList (Observable.java)
  at io.reactivex.internal.operators.observable.ObservableSingleSingle.subscribeActual (ObservableSingleSingle.java)
  at io.reactivex.Single.doOnSubscribe (Single.java)
  or                    .error (Single.java)
  or                    .flatMap (Single.java)
  or                    .just (Single.java)
  or                    .observeOn (Single.java)
  or                    .subscribe (Single.java)
  or                    .zip (Single.java)
  or                    .zipArray (Single.java)
  at io.reactivex.internal.operators.single.SingleMap.subscribeActual (SingleMap.java)
  at io.reactivex.Single.doOnSubscribe (Single.java)
  or                    .error (Single.java)
  or                    .flatMap (Single.java)
  or                    .just (Single.java)
  or                    .observeOn (Single.java)
  or                    .subscribe (Single.java)
  or                    .zip (Single.java)
  or                    .zipArray (Single.java)
  at io.reactivex.internal.operators.single.SingleZipArray.subscribeActual (SingleZipArray.java)
  at io.reactivex.Single.doOnSubscribe (Single.java)
  or                    .error (Single.java)
  or                    .flatMap (Single.java)
  or                    .just (Single.java)
  or                    .observeOn (Single.java)
  or                    .subscribe (Single.java)
  or                    .zip (Single.java)
  or                    .zipArray (Single.java)
  at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run (SingleSubscribeOn.java)
  at io.reactivex.Scheduler$DisposeTask.run (Scheduler.java)
  at io.reactivex.internal.schedulers.ScheduledRunnable.run (ScheduledRunnable.java)
  at io.reactivex.internal.schedulers.ScheduledRunnable.call (ScheduledRunnable.java)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201 (ScheduledThreadPoolExecutor.java:152)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:265)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
  at java.lang.Thread.run (Thread.java:841)
```

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/category/CategoryRepository.java
@@ -3,6 +3,7 @@ package openfoodfacts.github.scrachx.openfood.category;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 
+import android.util.Log;
 import io.reactivex.Single;
 import io.reactivex.schedulers.Schedulers;
 import openfoodfacts.github.scrachx.openfood.category.mapper.CategoryMapper;
@@ -27,6 +28,7 @@ public class CategoryRepository {
         return networkService.getCategories()
                 .map(categoryResponse -> mapper.fromNetwork(categoryResponse.getTags()))
                 .doOnSuccess(memoryCache::set)
+                .doOnError(throwable-> Log.w(CategoryRepository.class.getSimpleName(),"Can't get categories",throwable))
                 .subscribeOn(Schedulers.io());
     }
 }

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductIngredientsFragment.java
@@ -434,17 +434,21 @@ public class AddProductIngredientsFragment extends BaseFragment {
         if (requestCode == CropImage.CROP_IMAGE_ACTIVITY_REQUEST_CODE) {
             CropImage.ActivityResult result = CropImage.getActivityResult(data);
             if (resultCode == RESULT_OK) {
-                Uri resultUri = result.getUri();
-                imagePath = resultUri.getPath();
-                newImageSelected = true;
-                photoFile = new File((resultUri.getPath()));
-                ProductImage image = new ProductImage(code, INGREDIENTS, photoFile);
-                image.setFilePath(resultUri.getPath());
-                if (activity instanceof AddProductActivity) {
-                    ((AddProductActivity) activity).addToPhotoMap(image, 1);
+                if(result!=null && result.getUri().getPath()!=null) {
+                    Uri resultUri = result.getUri();
+                    imagePath = resultUri.getPath();
+                    newImageSelected = true;
+                    photoFile = new File((resultUri.getPath()));
+                    ProductImage image = new ProductImage(code, INGREDIENTS, photoFile);
+                    image.setFilePath(resultUri.getPath());
+                    if (activity instanceof AddProductActivity) {
+                        ((AddProductActivity) activity).addToPhotoMap(image, 1);
+                    }
+                }else{
+                    Log.e(AddProductIngredientsFragment.class.getSimpleName(),"Crop image result path not found "+result.getUri());
                 }
             } else if (resultCode == CropImage.CROP_IMAGE_ACTIVITY_RESULT_ERROR_CODE) {
-                Log.e("Crop image error", result.getError().toString());
+                Log.e(AddProductIngredientsFragment.class.getSimpleName(),"Crop image error: "+ result.getError().toString());
             }
         }
 

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/OfflineEditFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/OfflineEditFragment.java
@@ -250,7 +250,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                         @Override
                         public void onError(Throwable e) {
-                            Log.e(LOG_TAG, e.getMessage());
+                            Log.e(LOG_TAG, e.getMessage(),e);
                         }
                     });
         }
@@ -517,7 +517,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                                             @Override
                                             public void onError(Throwable e) {
-                                                Log.e(LOG_TAG, e.getMessage());
+                                                Log.e(LOG_TAG, e.getMessage(),e);
                                             }
                                         });
                             }
@@ -525,7 +525,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                         @Override
                         public void onError(Throwable e) {
-                            Log.e(LOG_TAG, e.getMessage());
+                            Log.e(LOG_TAG, e.getMessage(),e);
                         }
                     });
         } else {
@@ -613,7 +613,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                         @Override
                         public void onError(Throwable e) {
-                            Log.e(LOG_TAG, e.getMessage());
+                            Log.e(LOG_TAG, e.getMessage(),e);
                         }
                     });
         } else {
@@ -693,7 +693,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                                             @Override
                                             public void onError(Throwable e) {
-                                                Log.e(LOG_TAG, e.getMessage());
+                                                Log.e(LOG_TAG, e.getMessage(),e);
                                             }
                                         });
                             }
@@ -701,7 +701,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                         @Override
                         public void onError(Throwable e) {
-                            Log.e(LOG_TAG, e.getMessage());
+                            Log.e(LOG_TAG, e.getMessage(),e);
                         }
                     });
         } else {
@@ -769,7 +769,7 @@ public class OfflineEditFragment extends NavigationBaseFragment implements SaveL
 
                     @Override
                     public void onError(Throwable e) {
-                        Log.e(LOG_TAG, e.getMessage());
+                        Log.e(LOG_TAG, e.getMessage(),e);
                     }
                 });
     }

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java
@@ -438,6 +438,9 @@ public class Utils {
      * @return true if installed, false otherwise.
      */
     public static boolean isHardwareCameraInstalled(Context context) {
+        if(context==null){
+            return false;
+        }
         try {
             if (context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA)) {
                 return true;
@@ -520,6 +523,9 @@ public class Utils {
     }
 
     public static  boolean isUserLoggedIn(Context context){
+        if(context==null){
+            return false;
+        }
         final SharedPreferences settings = context.getSharedPreferences("login", 0);
         final String login = settings.getString("user", "");
         return StringUtils.isNotEmpty(login);

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AddProductActivity.java
@@ -1259,7 +1259,7 @@ public class AddProductActivity extends AppCompatActivity {
                         Snackbar.make(view, R.string.no_internet_unable_to_extract_ingredients, Snackbar.LENGTH_INDEFINITE)
                             .setAction(R.string.txt_try_again, v -> performOCR(code, imageField)).show();
                     } else {
-                        Log.i(this.getClass().getSimpleName(), e.getMessage());
+                        Log.i(this.getClass().getSimpleName(), e.getMessage(),e);
                         Toast.makeText(AddProductActivity.this, e.getMessage(), Toast.LENGTH_SHORT).show();
                     }
                 }

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/ContinuousScanActivity.java
@@ -353,9 +353,11 @@ public class ContinuousScanActivity extends android.support.v7.app.AppCompatActi
                                 fab_status.setVisibility(View.VISIBLE);
                                 quickView.setOnClickListener(v -> navigateToProductAddition(lastText));
                                 fab_status.setOnClickListener(v -> navigateToProductAddition(lastText));
+                            }else{
+                                Log.i(this.getClass().getSimpleName(), e.getMessage(),e);
                             }
                         } catch (Exception e1) {
-                            Log.i(this.getClass().getSimpleName(), e1.getMessage());
+                            Log.i(this.getClass().getSimpleName(), e1.getMessage(),e1);
                         }
                     }
                 });

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/LoadTaxonomiesService.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/LoadTaxonomiesService.java
@@ -6,6 +6,8 @@ import android.content.SharedPreferences;
 import android.support.annotation.Nullable;
 import java.util.Arrays;
 
+import android.util.Log;
+import android.widget.Toast;
 import io.reactivex.Completable;
 import io.reactivex.Single;
 import io.reactivex.android.schedulers.AndroidSchedulers;
@@ -55,14 +57,14 @@ public class LoadTaxonomiesService extends IntentService {
                         ).subscribeOn(Schedulers.computation())
                                 .subscribe(() -> {
                                     settings.edit().putLong(Utils.LAST_REFRESH_DATE, System.currentTimeMillis()).apply();
-                                }, Throwable::printStackTrace);
+                                }, throwable ->handleError(throwable));
 
                         return true;
                     })
                     .subscribeOn(Schedulers.io())
                     .observeOn(AndroidSchedulers.mainThread())
                     .toCompletable()
-                    .subscribe();
+                    .subscribe(()->{},throwable -> handleError(throwable));
         } else if (BuildConfig.FLAVOR.equals("obf")) {
             Single.zip(
                     productRepository.getLabels(true),
@@ -83,14 +85,14 @@ public class LoadTaxonomiesService extends IntentService {
                         ).subscribeOn(Schedulers.computation())
                                 .subscribe(() -> {
                                     settings.edit().putLong(Utils.LAST_REFRESH_DATE, System.currentTimeMillis()).apply();
-                                }, Throwable::printStackTrace);
+                                }, throwable ->handleError(throwable));
 
                         return true;
                     })
                     .subscribeOn(Schedulers.io())
                     .observeOn(AndroidSchedulers.mainThread())
                     .toCompletable()
-                    .subscribe();
+                    .subscribe(()->{},throwable ->handleError(throwable));
         } else if (BuildConfig.FLAVOR.equals("opf") || BuildConfig.FLAVOR.equals("opff")) {
             Single.zip(
                     productRepository.getTags(true),
@@ -110,8 +112,12 @@ public class LoadTaxonomiesService extends IntentService {
                     .subscribeOn(Schedulers.io())
                     .observeOn(AndroidSchedulers.mainThread())
                     .toCompletable()
-                    .subscribe();
+                    .subscribe(()->{},throwable ->handleError(throwable));
         }
     }
 
+    private void handleError(Throwable throwable) {
+        Toast.makeText(OFFApplication.getInstance(),R.string.errorWeb,Toast.LENGTH_LONG).show();
+        Log.e(LoadTaxonomiesService.class.getSimpleName(),"can't load products",throwable);
+    }
 }

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/PeriodAfterOpeningAutoCompleteAdapter.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/PeriodAfterOpeningAutoCompleteAdapter.java
@@ -60,7 +60,7 @@ public class PeriodAfterOpeningAutoCompleteAdapter extends ArrayAdapter<String> 
 
                                 @Override
                                 public void onError(Throwable e) {
-                                    Log.e(PeriodAfterOpeningAutoCompleteAdapter.class.getSimpleName(), e.getMessage());
+                                    Log.e(PeriodAfterOpeningAutoCompleteAdapter.class.getSimpleName(), e.getMessage(),e);
                                 }
                             });
 

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/ProductComparisonAdapter.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/ProductComparisonAdapter.java
@@ -393,7 +393,7 @@ public class ProductComparisonAdapter extends RecyclerView.Adapter<ProductCompar
                             ((TextView) v).setText(additivesBuilder.toString());
                             setMaxCardHeight();
                         }
-                            }, e -> Log.e(ProductComparisonAdapter.class.getSimpleName(),"loadAdditives",e))
+                    }, e -> Log.e(ProductComparisonAdapter.class.getSimpleName(), "loadAdditives", e))
             );
         }
     }

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductAttributeDetailsFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductAttributeDetailsFragment.java
@@ -1,6 +1,5 @@
 package openfoodfacts.github.scrachx.openfood.views.product;
 
-import android.content.Intent;
 import android.net.Uri;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
@@ -15,7 +14,6 @@ import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.TextView;
 import android.widget.Toast;
-
 import openfoodfacts.github.scrachx.openfood.R;
 import openfoodfacts.github.scrachx.openfood.models.*;
 import openfoodfacts.github.scrachx.openfood.utils.LocaleHelper;
@@ -26,11 +24,9 @@ import openfoodfacts.github.scrachx.openfood.views.ProductBrowsingListActivity;
 import openfoodfacts.github.scrachx.openfood.views.customtabs.CustomTabActivityHelper;
 import openfoodfacts.github.scrachx.openfood.views.customtabs.CustomTabsHelper;
 import openfoodfacts.github.scrachx.openfood.views.customtabs.WebViewFallback;
-
+import org.apache.commons.lang3.StringUtils;
 import org.json.JSONException;
 import org.json.JSONObject;
-
-import java.util.Locale;
 
 public class ProductAttributeDetailsFragment extends BottomSheetDialogFragment implements CustomTabActivityHelper.ConnectionCallback
 {
@@ -298,37 +294,34 @@ public class ProductAttributeDetailsFragment extends BottomSheetDialogFragment i
 	{
 	}
 
-	private String getDescription( JSONObject description )
-	{
+	private String getDescription(JSONObject description) {
 		String descriptionString = "";
+		if(description==null){
+			return descriptionString;
+		}
 		final String languageCode = LocaleHelper.getLanguage(OFFApplication.getInstance());
-		if( description.has( languageCode ) )
-		{
-			try
-			{
-				description = description.getJSONObject( languageCode );
-				descriptionString = description.getString( "value" );
-			}
-			catch( JSONException e )
-			{
-				e.printStackTrace();
+		if (languageCode!=null && description.has(languageCode)) {
+			try {
+				description = description.getJSONObject(languageCode);
+				if (description != null) {
+					descriptionString = description.getString("value");
+				}
+			} catch (JSONException e) {
+				Log.e(ProductAttributeDetailsFragment.class.getSimpleName(),e.getMessage(),e);
 			}
 		}
-		else if( description.has( "en" ) )
-		{
-			try
-			{
-				description = description.getJSONObject( "en" );
-				descriptionString = description.getString( "value" );
-			}
-			catch( JSONException e )
-			{
-				e.printStackTrace();
+		if (StringUtils.isEmpty(descriptionString) && description.has("en")) {
+			try {
+				description = description.getJSONObject("en");
+				if (description != null) {
+					descriptionString = description.getString("value");
+				}
+			} catch (JSONException e) {
+				Log.e(ProductAttributeDetailsFragment.class.getSimpleName(),e.getMessage(),e);
 			}
 		}
-		else
-		{
-			Log.i( "ProductActivity", "Result for description is not found in native or english language." );
+		if (StringUtils.isEmpty(descriptionString)) {
+			Log.i("ProductActivity", "Result for description is not found in native or english language.");
 		}
 		return descriptionString;
 	}

--- a/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java
+++ b/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/summary/SummaryProductFragment.java
@@ -54,7 +54,6 @@ import openfoodfacts.github.scrachx.openfood.views.customtabs.CustomTabActivityH
 import openfoodfacts.github.scrachx.openfood.views.customtabs.CustomTabsHelper;
 import openfoodfacts.github.scrachx.openfood.views.customtabs.WebViewFallback;
 import openfoodfacts.github.scrachx.openfood.views.product.ProductActivity;
-import openfoodfacts.github.scrachx.openfood.views.product.ProductFragment;
 import pl.aprilapps.easyphotopicker.DefaultCallback;
 import pl.aprilapps.easyphotopicker.EasyImage;
 

